staking-miner-playground/runtime/src/lib.rs

```diff
@@ -450,15 +461,21 @@ parameter_types! {
 
 	// miner configs
 	pub const ElectionUnsignedPriority: TransactionPriority = StakingUnsignedPriority::get() - 1u64;
-	pub MinerMaxWeight: Weight = Perbill::from_rational(8u32, 10) *
-		<Runtime as frame_system::Config>::BlockWeights::get()
-		.get(DispatchClass::Normal)
-		.max_total
-		.unwrap();
-	pub MinerMaxLength: u32 = Perbill::from_rational(8u32, 10) *
-		*(<<Runtime as frame_system::Config>::BlockLength as Get<limits::BlockLength>>::get())
-		.max
-		.get(DispatchClass::Normal);
+
+	// TODO(niklasad1): this is a hack to get the number of validators, candidates and nominators
```

Comment:
this question is really how configure these such that it doesn't too much such that `Incorrect number winners` doesn't occur

//cc @gpestana @kianenigma

Reply:
I recall configuring this with a goal like this: 

let's say you want 10 validators. provide 10 candidates, all of which will be winners. Given each of them a large number of evenly distributed voters, and then trim down. Technically this should not lead to an invalid solution. 

Ofc, you need to set the `MinerMaxWeight` such that we trim just a little bit, not too much. You can set that by feeding `MinerConfig::solution_weight()` repeatedly and finding a `v` for which we partially trim. 

https://github.com/paritytech/staking-miner-v2/blob/538/staking-miner-playground/runtime/src/lib.rs#L567

For example, If `MinerMaxweight` is 100, and you know `solution_weight(10, 100) == 120`, you want to find the biggest `x` for which `solution_weight(10, x) <= 100`.